packages/ui/src/components/card-select.tsx

```diff
@@ -0,0 +1,184 @@
+import { createContext, forwardRef, HTMLAttributes, ReactNode, useContext, useState } from 'react'
+
+import { cn } from '@utils/cn'
+import { cva } from 'class-variance-authority'
+
+import { Icon } from './icon'
+import { Logo } from './logo'
+
+type CardSelectType = 'single' | 'multiple'
+
+interface CardSelectRootProps<T> {
+  type: CardSelectType
+  name?: string
+  value?: T extends 'single' ? unknown : unknown[]
```

Comment:
can we use the same type inference pattern as Pagination to avoid using `unknown`?
something like

```ts
interface CardSingleSelectProps<T> {
  value: T,
  type: 'single'
}

interface CardMultiSelectProps<T> {
  value: Array<T>,
  type: 'multiple'
}

type CardSelectProps = CardSingleSelectProps | CardMultiSelectProps
```

It would vastly simplify the types for all the following interfaces and callbacks